### PR TITLE
Lots of code fixes & Yeet version checker

### PR DESCRIPTION
**Pull request by Alexdoru**
lots of code fixes, there is still more stuff that needs to be changes and also lots of unecessary reflection that should get removed

also I removed the version checker

### CODE REVIEW

File: src/main/java/logisticspipes/gui/hud/HUDPowerLevel.java [hunk 1]
```diff
@@ -60,10 +60,10 @@ public void drawTexturedModalRect(int par1, int par2, int par3, int par4, int pa
         float var8 = 0.00390625F;
         Tessellator var9 = Tessellator.instance;
         var9.startDrawingQuads();
-        var9.addVertexWithUV(par1 + 0, par2 + par6, 0, (par3 + 0) * var7, (par4 + par6) * var8);
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/gui/hud/HUDPowerLevel.java:63]: + 0 in guis means that this is the start point, allows to understand code more quick, so it should remain the same

File: src/main/java/logisticspipes/modules/ModuleCrafter.java [hunk 2]
```diff
@@ -570,11 +530,10 @@ public boolean isSatelliteConnected() {
             }
             for (final PipeItemsSatelliteLogistics satellite : PipeItemsSatelliteLogistics.AllSatellites) {
                 if (satellite.satelliteId == satelliteId) {
-                    CoreRoutedPipe satPipe = satellite;
-                    if (satPipe == null || satPipe.stillNeedReplace() || satPipe.getRouter() == null) {
+                    if (satellite == null || satellite.stillNeedReplace() || satellite.getRouter() == null) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/modules/ModuleCrafter.java:533]: why checking for null is after the checking for id?

File: src/main/java/logisticspipes/modules/ModuleCrafter.java [hunk 3]
```diff
@@ -591,11 +550,10 @@ public boolean isSatelliteConnected() {
                 }
                 for (final PipeItemsSatelliteLogistics satellite : PipeItemsSatelliteLogistics.AllSatellites) {
                     if (satellite.satelliteId == advancedSatelliteIdArray[i]) {
-                        CoreRoutedPipe satPipe = satellite;
-                        if (satPipe == null || satPipe.stillNeedReplace() || satPipe.getRouter() == null) {
+                        if (satellite == null || satellite.stillNeedReplace() || satellite.getRouter() == null) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/modules/ModuleCrafter.java:553]: why checking for null is after the checking for id?

File: src/main/java/logisticspipes/modules/ModuleCrafter.java [hunk 4]
```diff
@@ -788,21 +747,19 @@ public IRouter getSatelliteRouter(int x) {
         if (x == -1) {
             for (final PipeItemsSatelliteLogistics satellite : PipeItemsSatelliteLogistics.AllSatellites) {
                 if (satellite.satelliteId == satelliteId) {
-                    CoreRoutedPipe satPipe = satellite;
-                    if (satPipe == null || satPipe.stillNeedReplace() || satPipe.getRouter() == null) {
+                    if (satellite == null || satellite.stillNeedReplace() || satellite.getRouter() == null) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/modules/ModuleCrafter.java:750]: why checking for null is after the checking for id?

File: src/main/java/logisticspipes/modules/ModuleCrafter.java [hunk 5]
```diff
@@ -788,21 +747,19 @@ public IRouter getSatelliteRouter(int x) {
...
-                    return satPipe.getRouter();
+                    return satellite.getRouter();
                 }
             }
         } else {
             for (final PipeItemsSatelliteLogistics satellite : PipeItemsSatelliteLogistics.AllSatellites) {
                 if (satellite.satelliteId == advancedSatelliteIdArray[x]) {
-                    CoreRoutedPipe satPipe = satellite;
-                    if (satPipe == null || satPipe.stillNeedReplace() || satPipe.getRouter() == null) {
+                    if (satellite == null || satellite.stillNeedReplace() || satellite.getRouter() == null) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/modules/ModuleCrafter.java:759]: why checking for null is after the checking for id?

File: src/main/java/logisticspipes/modules/ModuleCrafter.java [hunk 6]
```diff
@@ -1191,21 +1148,19 @@ public IRouter getFluidSatelliteRouter(int x) {
         if (x == -1) {
             for (final PipeFluidSatellite satellite : PipeFluidSatellite.AllSatellites) {
                 if (satellite.satelliteId == liquidSatelliteId) {
-                    CoreRoutedPipe satPipe = satellite;
-                    if (satPipe == null || satPipe.stillNeedReplace() || satPipe.getRouter() == null) {
+                    if (satellite == null || satellite.stillNeedReplace() || satellite.getRouter() == null) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/modules/ModuleCrafter.java:1151]: why checking for null is after the checking for id?

File: src/main/java/logisticspipes/modules/ModuleCrafter.java [hunk 7]
```diff
@@ -1191,21 +1148,19 @@ public IRouter getFluidSatelliteRouter(int x) {
...
-                    return satPipe.getRouter();
+                    return satellite.getRouter();
                 }
             }
         } else {
             for (final PipeFluidSatellite satellite : PipeFluidSatellite.AllSatellites) {
                 if (satellite.satelliteId == liquidSatelliteIdArray[x]) {
-                    CoreRoutedPipe satPipe = satellite;
-                    if (satPipe == null || satPipe.stillNeedReplace() || satPipe.getRouter() == null) {
+                    if (satellite == null || satellite.stillNeedReplace() || satellite.getRouter() == null) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/modules/ModuleCrafter.java:1160]: why checking for null is after the checking for id?

File: src/main/java/logisticspipes/modules/ModuleCrafter.java [hunk 8]
```diff
@@ -1306,9 +1261,9 @@ public void enabledUpdateEntity() {
             if (getUpgradeManager().getCrafterCleanup() > 0) {
                 List<AdjacentTile> crafters = locateCrafters();
                 ItemStack extracted = null;
-                AdjacentTile tile = null;
-                for (Iterator<AdjacentTile> it = crafters.iterator(); it.hasNext(); ) {
-                    tile = it.next();
+                AdjacentTile tile;
+                for (AdjacentTile crafter : crafters) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/modules/ModuleCrafter.java:1265]: why do you need 'tile' here?

File: src/main/java/logisticspipes/modules/ModuleCrafter.java [hunk 9]
```diff
@@ -1356,8 +1311,8 @@ public void enabledUpdateEntity() {
             // retrieve the new crafted items
             ItemStack extracted = null;
             AdjacentTile tile = null;
-            for (Iterator<AdjacentTile> it = crafters.iterator(); it.hasNext(); ) {
-                tile = it.next();
+            for (AdjacentTile crafter : crafters) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/modules/ModuleCrafter.java:1314]: why do you need 'tile' here?

File: src/main/java/logisticspipes/network/GuiHandler.java [hunk 10]
```diff
@@ -76,7 +54,7 @@ public Object getServerGuiElement(int ID, EntityPlayer player, World world, fina
         if (ID < 110 && ID > 0) {
             switch (ID) {
                 case GuiIDs.GUI_FluidSupplier_ID:
-                    if (pipe == null || pipe.pipe == null || !(pipe.pipe instanceof PipeItemsFluidSupplier)) {
+                    if (pipe == null || !(pipe.pipe instanceof PipeItemsFluidSupplier)) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/network/GuiHandler.java:57]: redundant checking for null (and cases below - same), because instanceof already checks for null

File: src/main/java/logisticspipes/pipefxhandlers/PipeFXLaserPowerBall.java [hunk 11]
```diff
@@ -22,16 +22,16 @@ public class PipeFXLaserPowerBall extends EntityFX {
...
+    private final float random;
+    private final TileEntity tile;
 
     public PipeFXLaserPowerBall(World par1World, LPPosition pos, int color, TileEntity tile) {
         super(par1World, pos.getXD() + 0.5D, pos.getYD() + 0.5D, pos.getZD() + 0.5D, 0.0D, 0.0D, 0.0D);
         this.tile = tile;
         setSize(0.02F, 0.02F);
         particleRed = ((float) ((color & 0xff0000) >> 16)) / 0xff;
         particleGreen = ((float) ((color & 0x00ff00) >> 8)) / 0xff;
-        particleBlue = ((float) ((color & 0x0000ff) >> 0)) / 0xff;
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/pipefxhandlers/PipeFXLaserPowerBall.java:34]: needs to stay, since it improves readability

File: src/main/java/logisticspipes/pipefxhandlers/PipeFXLaserPowerBeam.java [hunk 12]
```diff
@@ -41,7 +41,7 @@ public PipeFXLaserPowerBeam(
         this.tile = tile;
         particleRed = ((float) ((color & 0xff0000) >> 16)) / 0xff;
         particleGreen = ((float) ((color & 0x00ff00) >> 8)) / 0xff;
-        particleBlue = ((float) ((color & 0x0000ff) >> 0)) / 0xff;
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/pipefxhandlers/PipeFXLaserPowerBeam.java:44]: needs to stay, since it improves readability

File: src/main/java/logisticspipes/pipes/PipeFluidSatellite.java [hunk 13]
```diff
@@ -243,10 +238,10 @@ protected void ensureAllSatelliteStatus() {
         if (MainProxy.isClient()) {
             return;
         }
-        if (satelliteId == 0 && PipeFluidSatellite.AllSatellites.contains(this)) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/pipes/PipeFluidSatellite.java:246]: why did you delete this check?

File: src/main/java/logisticspipes/pipes/PipeItemsSatelliteLogistics.java [hunk 14]
```diff
@@ -242,10 +240,10 @@ protected void ensureAllSatelliteStatus() {
         if (MainProxy.isClient()) {
             return;
         }
-        if (satelliteId == 0 && PipeItemsSatelliteLogistics.AllSatellites.contains(this)) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/pipes/PipeItemsSatelliteLogistics.java:245]: Why did you remove this check? (the same for two lines below)

File: src/main/java/logisticspipes/proxy/opencomputers/asm/BaseWrapperClass.java [hunk 15]
```diff
@@ -377,19 +370,18 @@ public void load(NBTTagCompound nbt) {
         if (type.equals("")) {
             return;
         }
-        if (type.equals("LPGlobalCCAccess")) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/proxy/opencomputers/asm/BaseWrapperClass.java:380]: 'if' is better than switch-case

File: src/main/java/logisticspipes/proxy/opencomputers/asm/BaseWrapperClass.java [hunk 16]
```diff
@@ -400,39 +392,43 @@ public Object call() throws Exception {
...
+                break;
+            }
+            case "CCItemIdentifierImplementation": {
+                ItemStack stack = ItemStack.loadItemStackFromNBT(nbt);
+                if (stack != null) {
+                    object = new CCItemIdentifierImplementation(ItemIdentifier.get(stack));
+                    checkType();
                 }
-            });
-        } else if (type.equals("CCItemIdentifierImplementation")) {
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/proxy/opencomputers/asm/BaseWrapperClass.java:405]: 'if' is better than switch-case

File: src/main/java/logisticspipes/renderer/LogisticsRenderPipe.java [hunk 17]
```diff
@@ -503,10 +502,10 @@ private void renderItem(IIcon par3Icon) {
         Tessellator tessellator = Tessellator.instance;
         tessellator.startDrawingQuads();
         tessellator.setNormal(0.0F, 1.0F, 0.0F);
-        tessellator.addVertexWithUV(par1 + 0, par2 + par5, zLevel, par3Icon.getMinU(), par3Icon.getMaxV());
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/renderer/LogisticsRenderPipe.java:506]: Zeros need to be left to improve readability of Vertex adding (the same for lines below)

File: src/main/java/logisticspipes/request/RequestTreeNode.java [hunk 18]
```diff
@@ -620,13 +605,10 @@ private static List<Pair<ICraftingTemplate, List<IFilter>>> getCrafters(
                     if (craftable != null) {
                         for (IFilter filter : r.filters) {
                             if (filter.isBlocked() == filter.isFilteredItem(craftable.getResultItem())
-                                    || filter.blockCrafting()) {
-                                continue;
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/request/RequestTreeNode.java:624]: Why was 'continue' deleted?

File: src/main/java/logisticspipes/ticks/ServerPacketBufferHandlerThread.java [hunk 19]
```diff
@@ -244,7 +229,7 @@ public void run() {
                         int size = ((ByteBufferForPlayer[0] & 255) << 24)
                                 + ((ByteBufferForPlayer[1] & 255) << 16)
                                 + ((ByteBufferForPlayer[2] & 255) << 8)
-                                + ((ByteBufferForPlayer[3] & 255) << 0);
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/ticks/ServerPacketBufferHandlerThread.java:247]: Zero needs to be left, since it improves readability

File: src/main/java/logisticspipes/utils/gui/extention/GuiExtentionController.java [hunk 20]
```diff
@@ -173,31 +172,28 @@ public void mouseClicked(int x, int y, int k) {
     }
 
     public void mouseOver(int i, int j) {
-        int x = i;
```
Comment (TimeConqueror) [inline src/main/java/logisticspipes/utils/gui/extention/GuiExtentionController.java:176]: Please rename parameters to x,y

File: src/main/java/logisticspipes/modules/ModuleCrafter.java (hunk 4 above)
Comment (Alexdoru) [reply src/main/java/logisticspipes/modules/ModuleCrafter.java:750]: it's the original code lol

File: src/main/java/logisticspipes/modules/ModuleCrafter.java (hunk 9 above)
Comment (Alexdoru) [reply src/main/java/logisticspipes/modules/ModuleCrafter.java:1314]: it's called lower in the code

File: src/main/java/logisticspipes/network/GuiHandler.java (hunk 10 above)
Comment (Alexdoru) [reply src/main/java/logisticspipes/network/GuiHandler.java:57]: it's not redundant since it's pipe and pipe.null

File: src/main/java/logisticspipes/pipes/PipeFluidSatellite.java (hunk 13 above)
Comment (Alexdoru) [reply src/main/java/logisticspipes/pipes/PipeFluidSatellite.java:246]: you don't need to check if an item is in a collection before removing it, same with adding

File: src/main/java/logisticspipes/request/RequestTreeNode.java (hunk 18 above)
Comment (Alexdoru) [reply src/main/java/logisticspipes/request/RequestTreeNode.java:624]: because it's the last statement of the loop, question is : why is there a loop with just an empty if statement ?

File: src/main/java/logisticspipes/gui/hud/HUDPowerLevel.java (hunk 1 above)
Comment (Alexdoru) [reply src/main/java/logisticspipes/gui/hud/HUDPowerLevel.java:63]: personally I think it makes the code look terrible, but sure

**Review by TimeConqueror — APPROVED** [review]
